dspace-api/src/main/java/org/dspace/app/orcid/client/OrcidClientImpl.java

```diff
@@ -0,0 +1,164 @@
+        params.add(new BasicNameValuePair("client_secret", orcidConfiguration.getClientSecret()));
+
+        HttpUriRequest httpUriRequest = RequestBuilder.post(orcidConfiguration.getTokenEndpointUrl())
+            .addHeader("Content-Type", "application/x-www-form-urlencoded")
+            .addHeader("Accept", "application/json")
+            .setEntity(new UrlEncodedFormEntity(params, Charset.defaultCharset()))
+            .build();
+
+        return executeAndParseJson(httpUriRequest, OrcidTokenResponseDTO.class);
+
+    }
+
+    @Override
+    public Person getPerson(String accessToken, String orcid) {
+        HttpUriRequest httpUriRequest = buildGetUriRequest(accessToken, "/" + orcid + "/person");
+        return executeAndUnmarshall(httpUriRequest, false, Person.class);
+    }
+
+    private HttpUriRequest buildGetUriRequest(String accessToken, String relativePath) {
+        return get(orcidConfiguration.getApiUrl() + relativePath.trim())
+            .addHeader("Content-Type", "application/x-www-form-urlencoded")
+            .addHeader("Authorization", "Bearer " + accessToken)
+            .build();
+    }
+
+    private <T> T executeAndParseJson(HttpUriRequest httpUriRequest, Class<T> clazz) {
+
+        HttpClient client = HttpClientBuilder.create().build();
+
+        return executeAndReturns(() -> {
+
+            HttpResponse response = client.execute(httpUriRequest);
+
+            if (isNotSuccessfull(response)) {
+                throw new OrcidClientException(getStatusCode(response), formatErrorMessage(response));
+            }
+
+            return objectMapper.readValue(response.getEntity().getContent(), clazz);
+
+        });
+
+    }
+
+    private <T> T executeAndUnmarshall(HttpUriRequest httpUriRequest, boolean handleNotFoundAsNull, Class<T> clazz) {
+
+        HttpClient client = HttpClientBuilder.create().build();
+
+        return executeAndReturns(() -> {
+
+            HttpResponse response = client.execute(httpUriRequest);
+
+            if (handleNotFoundAsNull && isNotFound(response)) {
+                return null;
+            }
+
+            if (isNotSuccessfull(response)) {
+                throw new OrcidClientException(getStatusCode(response), formatErrorMessage(response));
+            }
+
+            return unmarshall(response.getEntity(), clazz);
+
+        });
+    }
+
+    private <T> T executeAndReturns(ThrowingSupplier<T, Exception> supplier) {
+        try {
+            return supplier.get();
+        } catch (OrcidClientException ex) {
+            throw ex;
+        } catch (Exception ex) {
+            throw new OrcidClientException(ex);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T unmarshall(HttpEntity entity, Class<T> clazz) throws Exception {
+        InputStream content = entity.getContent();
+        JAXBContext jaxbContext = JAXBContext.newInstance(clazz);
+        Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
```

Comment:
To solve the other LGTM warning, we need to disable DTDs when parsing this content. I'd recommend we solve it (if possible to do so with no side effects)...even though ORCID is "trusted", we should still protect ourselves just in case.

I think the following (based on https://stackoverflow.com/a/68573613/3750035) will solve the warning:
```
JAXBContext jaxbContext = JAXBContext.newInstance(clazz);
XMLInputFactory xif = XMLInputFactory.newFactory();
xif.setProperty(XMLInputFactory.SUPPORT_DTD, false);
XMLStreamReader xsr = xif.createXMLStreamReader(content);
Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
return (T) unmarshaller.unmarshal(xsr);
```